flowpipe/graph.py

```diff
@@ -127,9 +127,9 @@ def deserialize(data):
         graph._nodes = []
         for node in data['nodes']:
             graph._nodes.append(INode.deserialize(node))
+        nodes = {n.identifier: n for n in graph.nodes}
```

Comment:
I believe you can use this dict also to replace the loops in ll. 135/136 and 141/142. As in
```python
for identifier, plug in input_['connections'].items():
    upstream = nodes[identifier]
```

Reply:
Thanks for catching that, should be addressed now